backend/src/clients/cognito.ts

```diff
@@ -0,0 +1,37 @@
+import { CognitoIdentityProviderClient, ListUsersCommand } from '@aws-sdk/client-cognito-identity-provider'
+
+import { UserInformation } from '../connectors/v2/authentication/Base.js'
+import config from '../utils/v2/config.js'
+/**
+ * Make this into a client function queryEmails that returns an array of emails
+ * Handle errors like missing userpool Id
+ */
+export async function listUsers(query: string) {
+  const dnName = config.oauth.cognito.userIdAttribute
+  const client = new CognitoIdentityProviderClient(config.oauth.cognito.identityProviderClient)
+
+  const command = new ListUsersCommand({
+    UserPoolId: config.oauth.cognito.userPoolId,
+    Filter: `"${dnName}"^="${query}"`,
```

Comment:
Add a parameter to change this to absolute match, otherwise emails that share prefixes will break.